### DISCUSSION TITLE
Export spaCy models for use in Java environments?

### DISCUSSION THREAD

**Discussion by matanox** [explosion/spaCy#2466] [New Features & Project Ideas]
## Feature description
It would be nice being able to export NER models (or any other spaCy trained models) so that they can be run by Java code, e.g. by one's own model runner. Is this already possible?

**Comment by ines** [comment]
That doesn't exist yet, but it'd be a nice project idea for someone from the community 🙂 I'm not sure how the specifics of this would look in Java, but ideally, the solution could be a wrapper for loading the existing model data? This would allow the users to download the same files for both Python and Java, and make it easier to keep them consistent and in sync.

**Comment by matanox** [comment]
@ines Exactly. Is the model data structure best described by some specific part of the code base?

**Comment by mortezakz** [comment]
Ah... if only this existed :-)
